bears/vcs/mercurial/HgCommitBear.py

```diff
@@ -0,0 +1,210 @@
+import re
+import shutil
+import os
+
+from urllib.parse import urlparse
+from coala_utils.ContextManagers import change_directory
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+from bears.vcs.CommitBear import _CommitBear
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+class HgCommitBear(_CommitBear):
+    LANGUAGES = {'Hg'}
+    SUPPORTED_HOST_KEYWORD_REGEX = {
+        'bitbucket': (r'[Cc]los(?:e[sd]?|ing)'
+                      r'|[Rr]esolv(?:e[sd]?|ing)'
+                      r'|[Ff]ix(?:e[sd]|ing)?')
+    }
+
+    @classmethod
+    def get_shortlog_checks_metadata(cls):
```

Comment:
why cant these methods move into the new base class?

Reply:
I also though of moving all these methods into the base class.
But what was happening is as follows:
The get_issue_checks_metadata function relies on the check_issue_reference function
The latter is specific to HgCommitBear, whereas all other metadata related functions are more relevant to the base class.
So when I moved all of them into base class and created an abstract function as a placeholder for check_issue_reference and redefined it in the inherited class, there was an issue in getting the code coverage to 100%
I couldn't get how to create a test for the abstract function, since it is in a class that cannot be instantiated.

Any ideas/pointers on this will be much helpful :smile:

Reply:
show me a build with <100% coverage, and i'll try to help